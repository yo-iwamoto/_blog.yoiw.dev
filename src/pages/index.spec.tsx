import Page from './index.page';
import { renderNextPage } from '@/__tests__/util';

describe('[page] /', () => {
  it('renders / unchanged', () => {
    const { container } = renderNextPage(
      <Page
        status='ok'
        data={{
          pages: [
            {
              id: 'id',
              properties: {
                title: 'title',
                tags: [
                  {
                    id: 'id',
                    name: 'name',
                    color: 'gray',
                  },
                ],
                slug: 'slug',
                is_public: true,
                created_at: 'yyyy/mm/dd',
                updated_at: 'yyyy/mm/dd',
              },
            },
          ],
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
