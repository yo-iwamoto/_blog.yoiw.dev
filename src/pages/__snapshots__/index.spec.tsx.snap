// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[page] / renders / unchanged 1`] = `
<div>
  <div
    class="font-sans text-slate-800"
  >
    <div
      class="mx-4 my-6"
    >
      <div
        class="mx-auto max-w-5xl px-2"
      >
        <header
          class="flex items-center justify-between"
        >
          <a
            href="/"
          >
            <p
              class="font-mono text-lg font-bold"
            >
              blog.yoiw.dev
            </p>
          </a>
          <form
            class="flex h-12 items-center justify-end gap-4"
          >
            <input
              class="w-auto max-w-[128px] border-b-2 border-b-slate-500 py-1 px-2 transition-opacity focus:outline-blue-300 sm:w-56 sm:max-w-none md:w-80 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 opacity-0"
              type="text"
              value=""
            />
            <svg
              class="cursor-pointer text-xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </form>
        </header>
        <div
          class="mt-4"
        >
          <div
            class="flex flex-col gap-4"
          >
            <a
              href="/entries/slug"
            >
              <div
                class="cursor-pointer rounded-lg bg-slate-50 p-4 shadow-md transition-all hover:opacity-80 hover:shadow-lg"
              >
                <h1
                  class="text-lg font-bold sm:text-xl md:text-2xl"
                >
                  title
                </h1>
                <p
                  class="mb-1 text-sm text-slate-500 md:text-lg"
                >
                  yyyy/mm/dd
                </p>
                <div
                  class="flex gap-2"
                >
                  <span
                    class="text-slate-700 before:pr-0.5 before:content-[\\"#\\"]"
                  >
                    name
                  </span>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
